app/code/Magento/Search/Test/Mftf/Test/StorefrontVerifySearchTermEntityRedirectTest.xml

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StorefrontVerifySearchTermEntityRedirectTest">
+        <annotations>
+            <stories value="Search Term Redirect"/>
+            <title value="Create Search Term Entity With Redirect. Check How Redirect is Working on Storefront"/>
+            <description value="Storefront search by created search term with redirect. Verifying if created redirect is working"/>
+        </annotations>
+
+        <before>
+            <!-- Login As Admin User -->
+            <actionGroup ref="LoginAsAdmin" stepKey="loginAsAdmin"/>
+            <!-- Navigate To Marketing Search Terms Grid -->
+            <actionGroup ref="AdminNavigateMenuActionGroup" stepKey="navigateToSearchTermPage">
+                <argument name="menuUiId" value="{{AdminMenuMarketing.dataUiId}}"/>
+                <argument name="submenuUiId" value="{{AdminMenuMarketingSEOAndSearchSearchTerms.dataUiId}}"/>
+            </actionGroup>
+            <!-- Create Custom Search Term With Redirect -->
+            <actionGroup ref="AdminCreateNewSearchTermEntityActionGroup" stepKey="createSearchTerm">
+                <argument name="query" value="{{SearchTerm.query_text}}"/>
+                <argument name="store" value="{{SearchTerm.store_id}}"/>
+                <argument name="redirectUrl" value="{{SearchTerm.redirect}}"/>
+            </actionGroup>
+        </before>
+        <after>
+            <amOnPage url="{{AdminCatalogSearchTermIndexPage.url}}" stepKey="navigateToSearchTermPage"/>
+            <actionGroup ref="AdminSearchTermFilterBySearchQueryActionGroup" stepKey="findCreatedTerm">
+                <argument name="searchQuery" value="{{SearchTerm.query_text}}"/>
+            </actionGroup>
+            <actionGroup ref="AdminDeleteSearchTermActionGroup" stepKey="deleteCreatedSearchTerm"/>
```

Comment:
Could you create `AdminDeleteSearchTermByQueryActionGroup` that actually extends `AdminSearchTermFilterBySearchQueryActionGroup` and adds `<amOnPage url="{{AdminCatalogSearchTermIndexPage.url}}" stepKey="navigateToSearchTermPage"/>` before first step and `<actionGroup ref="AdminDeleteSearchTermActionGroup" stepKey="deleteCreatedSearchTerm"/>` after last step? :-)

I'm pretty sure that other tests related with SearchTerms will benefit from that change.